### PR TITLE
baseline time tracer code

### PR DESCRIPTION
**Pull request by havardpe**
@vekterli please review
@arnej27959 @bjorncs FYI

### CODE REVIEW

File: vespalib/src/vespa/vespalib/test/time_tracer.h [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ * Typically, timing information is collected while a unit test is
+ * running. Sampling can be performed in the test code itself to
+ * identify which parts take more time. For more advanced analysis,
+ * critical parts of the code being tested can also be modified to
+ * collect time samples (this is similar to the ancient fprintf
+ * debugging technique).
+ *
+ * In order to collect timing information, the client code uses the
+ * TT_Tag and TT_Sample classes. The TT_Tag class represents a thing
+ * that can happen. It should always be constructed up font. The
```
Comment (vekterli) [inline vespalib/src/vespa/vespalib/test/time_tracer.h]: s/font/front/

File: vespalib/src/vespa/vespalib/test/time_tracer.cpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    : _lock(),
+      _state_list(),
+      _tags()
+{
+}
+
+uint32_t
+TimeTracer::get_tag_id(const vespalib::string &tag_name)
+{
+    auto guard = std::lock_guard(_lock);
```
Comment (vekterli) [inline vespalib/src/vespa/vespalib/test/time_tracer.cpp]: Purely out of interest, why this instead of `std::lock_guard guard(_lock);`?

**Review by vekterli — APPROVED** [review]
:+1: just a minor question and a very minor typo

File: vespalib/src/vespa/vespalib/test/time_tracer.cpp (hunk 2 above)
Comment (havardpe) [reply vespalib/src/vespa/vespalib/test/time_tracer.cpp]: fixed (did not remember the best way to enable auto-deduction)

File: vespalib/src/vespa/vespalib/test/time_tracer.h (hunk 1 above)
Comment (havardpe) [reply vespalib/src/vespa/vespalib/test/time_tracer.h]: fixed

**Comment by havardpe** [comment]
PTAL

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**